frontend/src/features/public-form/components/StatusTrackerPage/TimelineRunSteps.tsx

```diff
@@ -0,0 +1,146 @@
+import { MdCheck, MdClose } from 'react-icons/md'
+import {
+  BackgroundProps,
+  Box,
+  Circle,
+  Divider,
+  Flex,
+  Icon,
+  Stack,
+  Text,
+} from '@chakra-ui/react'
+import { format } from 'date-fns'
+
+import { StepData, WorkflowStatus } from '~shared/types'
+
+const statusColor: Record<WorkflowStatus, BackgroundProps['bg']> = {
+  [WorkflowStatus.PENDING]: '#E5E9F8', // grey
+  [WorkflowStatus.COMPLETED]: '#05CC9A', // green
+  [WorkflowStatus.APPROVED]: '#05CC9A',
+  [WorkflowStatus.REJECTED]: '#C03434', // red
+}
+
+const statusTextMap: Record<WorkflowStatus, string | undefined> = {
+  [WorkflowStatus.APPROVED]: 'Approved',
+  [WorkflowStatus.REJECTED]: 'Not Approved',
+  [WorkflowStatus.PENDING]: undefined,
+  [WorkflowStatus.COMPLETED]: undefined,
+}
+
+const StatusIcon = ({
+  workflowStatus,
+  stepNumber,
+  isCurrentPendingStep,
+}: {
+  workflowStatus: WorkflowStatus
+  stepNumber: number
+  isCurrentPendingStep: boolean | undefined
+}): JSX.Element => {
+  if (isCurrentPendingStep) {
+    return (
+      <Circle size="2rem" border="2px" borderColor="warning.500" bg="yellow.50">
+        <Text textStyle="caption-2">{stepNumber}</Text>
+      </Circle>
+    )
+  }
+
+  switch (workflowStatus) {
+    case WorkflowStatus.COMPLETED:
+    case WorkflowStatus.APPROVED:
+      return (
+        <Circle size="2rem" bg={statusColor[workflowStatus]}>
+          <Icon as={MdCheck} color="#F5F6F8" boxSize="1rem" />
+        </Circle>
+      )
+    case WorkflowStatus.PENDING:
+      return (
+        <Circle
+          size="2rem"
+          border="2px solid"
+          borderColor={statusColor[workflowStatus]}
+        >
+          <Text textStyle="caption-2">{stepNumber}</Text>
+        </Circle>
+      )
+    case WorkflowStatus.REJECTED:
+      return (
+        <Circle size="2rem" bg={statusColor[workflowStatus]}>
+          <Icon as={MdClose} color="#F5F6F8" boxSize="1rem" />
+        </Circle>
+      )
```

Comment:
this `#F5F6F8` colour is used just twice, so there's no need to extract it out (saves two lines of code)
but if it gets repeated a third time, might wanna make it a token :)